src/System.Device.Gpio/System/Device/Spi/Devices/UnixSpiDevice.cs

```diff
@@ -182,9 +203,22 @@ public override unsafe void Write(ReadOnlySpan<byte> buffer)
         {
             Initialize();
 
-            fixed (byte* dataPtr = buffer)
+            if (_isInverted)
             {
-                Transfer(dataPtr, null, buffer.Length);
+                Span<byte> toSend = stackalloc byte[buffer.Length];
```

Comment:
You should not `stackalloc` variable buffer length as you can easily stackoverflow here - instead cap stackalloc to some number of bytes and affter that do regular allocation. See this as an example:
https://github.com/dotnet/runtime/blob/1d9e50cb4735df46d3de0cee5791e97295eaf588/src/libraries/System.Text.Json/src/System/Text/Json/Reader/JsonReaderHelper.Unescaping.cs#L17-L19

If you can use ArrayPool that's good, if not then regular `new byte[...]` is fine

Reply:
The number of bytes typically sent is very small. Isn't there a length limit for SPI transfers?

Reply:
I'll go for a normal new byte[], no need of a Span anyway, what is sent if a normal buffer then.

Reply:
Reworked all of them as it's only byte[] based. Simplified and make it more clear.